### PR TITLE
Callhome - new add-on

### PR DESCRIPTION
**Pull request by psiinon**
To be used by the core CFU code, the Quick Start news and telemetry (in time).

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Is the duplicate OS and detection code going to be removed from core eventually?

**Comment by psiinon** [comment]
I was thinking that it would stay in core and be removed from this add-on ... but it could be the other way around I suppose..

**Comment by kingthorin** [comment]
Whichever, as long as we aren’t trying to keep two in sync long term 😉

**Comment by psiinon** [comment]
Thats what this comment was supposed to implie: `// Copied from the core - can be removed from 2.12.0 .. unless more changes are mode after that.`

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Great, I had thought that was just about the car or whatever near where it was located. But all good, we’re aligned.

**Comment by thc202** [comment]
If core does not need the specific OS detection (which does not seem to be the case, at least for now)  it might be better move to the add-on instead (if it's generally useful to all add-ons it could be moved to commonlib), that way we can add support for more OSes without requiring a full release.

**Comment by psiinon** [comment]
Updated based on feedback and to implement the Supplier interface - used by core: https://github.com/zaproxy/zaproxy/pull/6932

**Review by thc202 — APPROVED** [review]

File: addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        json.put(
+                "osVersion",
+                System.getProperty("os.name") + " : " + System.getProperty("os.version"));
+        json.put("javaVersion", System.getProperty("java.version"));
+        json.put("zapType", ZAP.getProcessType().name());
+        json.put("container", Constant.isInContainer() ? Constant.getContainerName() : "");
+        return json;
+    }
+
+    public ZapXmlConfiguration getCheckForUpdatesData()
```
Comment (thc202) [inline addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java:138]: This could be private now?

Comment (psiinon) [reply addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java:138]: I would prefer to keep it public. The news equivalent will be public (to be added once this has been merged) and you never know when someone will come up with a good usecase for usinging this somewhere else...

Comment (thc202) [reply addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java:138]: OK.

File: addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            } else if (Constant.isMacOsX()) {
+                os = OS.MAC_OS;
+            } else {
+                os = OS.UNKNOWN;
+            }
+        }
+        return os;
+    }
+
+    // XXX: Use Constant.isBackBox() from 2.12.0
```
Comment (kingthorin) [inline addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java:199]: I believe this comment can go? It's being dropped from core in the other PR: https://github.com/zaproxy/zaproxy/pull/6932/

Comment (psiinon) [reply addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java:197]: No, it should still be there :/

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**